src/sql/run/run.py

```diff
@@ -0,0 +1,142 @@
+import sqlparse
+
+from sql import exceptions, display
+from sql.run.resultset import ResultSet
+from sql.run.pgspecial import handle_postgres_special
+
+
+# some dialects have autocommit specific dialects break when commit is used
+_COMMIT_BLACKLIST_DIALECTS = (
+    "athena",
+    "bigquery",
+    "clickhouse",
+    "ingres",
+    "mssql",
+    "teradata",
+    "vertica",
+)
+
+
+def run_statements(conn, sql, config):
+    """
+    Run a SQL query (supports running multiple SQL statements) with the given
+    connection. This is the function that's called when executing SQL magic.
+
+    Parameters
+    ----------
+    conn : sql.connection.AbstractConnection
+        The connection to use
+
+    sql : str
+        SQL query to execution
+
+    config
+        Configuration object
+    """
+    if not sql.strip():
+        return "Connected: %s" % conn.name
+
+    for statement in sqlparse.split(sql):
+        first_word = sql.strip().split()[0].lower()
+
+        # TODO: checking for with isn't the best idea because it doesn't guarantee
+        # that the final one is a select statement
+        first_word_statement = statement.strip().split()[0].lower()
+
+        # in duckdb db "from TABLE_NAME" is valid
+        is_select = first_word_statement in {"select", "with", "from"}
+
+        if first_word == "begin":
+            raise exceptions.RuntimeError("JupySQL does not support transactions")
```

Comment:
Just realize we don't have a test case to cover this condition, we can cover it in next PR

Reply:
we have a test. test_run.py, test named: test_sql_starts_with_begin